File: solutions/easy/1346.Check_If_N_and_Its_Double_Exist/1346.Check_If_N_and_Its_Double_Exist.py
from ast import List


class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        arr_set = set(arr)
        for i in range(len(arr)):
            if arr[i] * 2 in arr_set and arr[i] != 0:
                return True
            elif arr[i] == 0 and arr.count(0) > 1:
                return True
        return False

File: solutions/easy/1346.Check_If_N_and_Its_Double_Exist/1346.Check_If_N_and_Its_Double_Exist.ts
function checkIfExist(arr: number[]): boolean {
  arr.sort((a, b) => a - b);
  for (let i = 0; i < arr.length - 1; i++) {
    if (arr[i] * 2 === arr[i + 1] || arr[i] === arr[i + 1] * 2) {
      return true;
    }
  }
  return false;
};

//or this solution

function checkIfExistSecond(arr: number[]): boolean {
  const set = new Set();
  for (const num of arr) {
    if (set.has(num * 2) || set.has(num / 2)) {
      return true;
    }
    set.add(num);
  }
  return false;
};